### PR TITLE
fix: update binlog index memory uasge before loading segments

### PR DESCRIPTION
**Pull request by cqy123456**
issue: #27678 
when interimIndex = true, memory predict should be update with the memory usage of binlog index build process.

### CODE REVIEW

**Comment by cqy123456** [comment]
rerun ut

File: internal/querynodev2/segments/segment_loader.go [hunk 1]
```diff
@@ -50,7 +50,8 @@ import (
 )
 
 const (
-	UsedDiskMemoryRatio = 4
+	UsedDiskMemoryRatio     = 4
+	BinlogIndexExpandFactor = 1.15
 )
```
Comment (congqixia) [inline internal/querynodev2/segments/segment_loader.go:55]: please add these magic number into paramtable instead of using const here

**Review by congqixia — CHANGES_REQUESTED** [review]

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**